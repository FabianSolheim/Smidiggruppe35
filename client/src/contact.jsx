import { useState } from "react";
import React from "react";
import Button from "@mui/material/Button";
import { TextField } from "@mui/material";

export function Contact() {
  return (
    <div>
      <h1>Contact</h1>
      <p>
        Reach out if you want to collaborate or know more about our services or
        just to connect
      </p>
        <ContactForm/>
    </div>
  );
}

function ContactForm() {
  const [name, setName] = useState("");
  const [companyName, setCompanyName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");

  return (
    <form>
      <div>
        <TextField required label={"Name"} variant={"standard"} />
          {/*
        <input
          id={"name"}
          name={name}
          onChange={(event) => setName(event.target.value)}
        />*/}
      </div>
      <div>
        <TextField required label={"Co. Name"} variant={"standard"} />
          {/*
        <input
          name={companyName}
          onChange={(event) => setCompanyName(event.target.value)}
        />*/}
      </div>
      <div>
        <TextField required label={"Email"} variant={"standard"}/>

          {/*
          <input
            name={email}
            onChange={(event) => setEmail(event.target.value)}
          />*/}
      </div>
      <div>
        <TextField required label="Message" multiline rows={4} />
          {/*
        <textarea
          name={message}
          onChange={(event) => setMessage(event.target.value)}
        />*/}
      </div>
      <div>
        <Button variant={"contained"}>Submit</Button>
      </div>
    </form>
  );
}
