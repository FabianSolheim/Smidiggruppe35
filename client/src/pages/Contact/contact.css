.contactContainer {
    margin-top: 200px;
}

.contactHeader {
    margin-bottom: 50px;
}

.formInfo {
    margin-bottom: 15px;
}

.textHighlight{
    font-weight: 700;
    color: #034f7a;
}

.contactInfo {
    margin-bottom: 30px;
    justify-content: center;
    text-align: center;
}

.contactInfoHover:hover{
    color: #034f7a !important;
}