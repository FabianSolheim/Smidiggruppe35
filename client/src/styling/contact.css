.contactContainer {
    margin-top: 200px;
}

.contactHeader {
    margin-bottom: 50px;
}

.formInfo {
    margin-bottom: 15px;
}

.contactInfo {
    margin-bottom: 30px;
    justify-content: center;
    text-align: center;
}