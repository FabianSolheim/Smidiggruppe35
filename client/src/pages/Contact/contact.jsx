import { useState } from "react";
import React from "react";
import { Box, InputAdornment, TextField } from "@mui/material";
import PersonIcon from "@mui/icons-material/Person";
import EmailIcon from "@mui/icons-material/Email";
import BusinessIcon from "@mui/icons-material/Business";
import PhoneIcon from "@mui/icons-material/Phone";
import LocationOnIcon from "@mui/icons-material/LocationOn";
import Grid from "@mui/material/Grid";
import "./contact.css";
import CustomButton from "../../components/CustomButton/CustomButton";
import { HeadlineWithUnderline } from "../../components/HeadlineWithUnderline/headlineWithUnderline";

export function Contact() {
  return (
    <Box mt={10} mb={20}>
      <Grid container justifyContent="center" alignItems="center">
        <header className="contactHeader">
          <Grid item xs={12}>
            {HeadlineWithUnderline("Contact", 120, "#034F7A", 20)}
            <p style={{ maxWidth: 300, textAlign: "center", marginBottom: 30 }}>
              Reach out if you want to <span className={"textHighlight"}>collaborate</span> or know more about our
              services, or just to <span className={"textHighlight"}>connect</span>.
            </p>
          </Grid>
        </header>
        <Grid container gap={20} alignItems="center" justifyContent="center">
          <Grid item>
            <ContactForm color={"primary"} />
          </Grid>

          <Grid item>
            <div className="contactInfo">
              <PhoneIcon sx={{ fontSize: 50 }} color={"primary"} />
              <p>+47 92 45 26 10</p>
            </div>
            <div className="contactInfo">
              <EmailIcon sx={{ fontSize: 50 }} color={"primary"} />
              <p>contact@melioraimpact.no</p>
            </div>
            <div className="contactInfo">
              <LocationOnIcon sx={{ fontSize: 50 }} color={"primary"} />
              <p>Peder Ankers plass 49</p>
              <p>0301 Oslo</p>
            </div>
          </Grid>
        </Grid>
      </Grid>
    </Box>
  );
}

function ContactForm() {
  //om man skal få den til å lagre informasjonen man skriver inn
  const [name, setName] = useState("");
  const [companyName, setCompanyName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");
  /*const toast = useToast();*/

  return (
    <form>
      <div className="formInfo">
        <TextField
          required
          color={"primary"}
          label={"Contact person"}
          sx={{ width: 330 }}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <PersonIcon sx={{fontSize: 40}} color={"primary"} />
              </InputAdornment>
            ),
          }}
          variant={"standard"}
        />
      </div>
      <div className="formInfo">
        <TextField
          required
          label={"Co. Name"}
          sx={{ width: 330 }}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <BusinessIcon sx={{fontSize: 40}} color={"primary"} />
              </InputAdornment>
            ),
          }}
          variant={"standard"}
        />
      </div>
      <div className="formInfo">
        <TextField
          required
          label={"Email"}
          sx={{ width: 330 }}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <EmailIcon sx={{fontSize: 40}} color={"primary"} />
              </InputAdornment>
            ),
          }}
          variant={"standard"}
        />
      </div>
      <div className="formInfo">
        <TextField
          required
          label="Message"
          sx={{ width: 330 }}
          multiline
          rows={4}
        />
      </div>
      <div align="center">
        <CustomButton
          variant="contained"
          to={"/contact"}
          /*
          Toast with chakra
          onClick={() =>
            toast({
              title: "Message sent.",
              description: "Thank you for reaching out to us.",
              status: "success",
              duration: 9000,
              isClosable: true,
            })
          }*/
        >
          Submit
        </CustomButton>
      </div>
    </form>
  );
}
