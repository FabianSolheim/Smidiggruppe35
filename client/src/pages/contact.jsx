import { useState } from "react";
import React from "react";
import { Box, InputAdornment, TextField } from "@mui/material";
import PersonIcon from "@mui/icons-material/Person";
import EmailIcon from "@mui/icons-material/Email";
import BusinessIcon from "@mui/icons-material/Business";
import PhoneIcon from "@mui/icons-material/Phone";
import LocationOnIcon from "@mui/icons-material/LocationOn";
import Grid from "@mui/material/Grid";
import "../styling/contact.css";
import Typography from "@mui/material/Typography";
import CustomButton from "../components/CustomButton/CustomButton";
import { HeadlineWithUnderline } from "../components/HeadlineWithUnderline/headlineWithUnderline";
import { useToast } from "@chakra-ui/react";

export function Contact() {
  return (
    <Box mt={10} mb={20}>
      <Grid container justifyContent="center" alignItems="center">
        <header className="contactHeader">
          <Grid item xs={12}>
            {HeadlineWithUnderline("Contact", 120, "#034F7A", 10)}
            <Typography paragraph>
              Reach out if you want to collaborate or know more about our
              services, or just to connect
            </Typography>
          </Grid>
        </header>
        <Grid container gap={12} alignItems="center" justifyContent="center">
          <Grid item>
            <ContactForm color={"primary"} />
          </Grid>

          <Grid item>
            <div className="contactInfo">
              <PhoneIcon color={"primary"} />
              <Typography paragraph>+47 92 45 26 10</Typography>
            </div>
            <div className="contactInfo">
              <EmailIcon color={"primary"} />
              <Typography paragraph>contact@melioraimpact.no</Typography>
            </div>
            <div className="contactInfo">
              <LocationOnIcon color={"primary"} />
              <Typography paragraph>Peder Ankers plass 49 0301 Oslo</Typography>
            </div>
          </Grid>
        </Grid>
      </Grid>
    </Box>
  );
}

function ContactForm() {
  //om man skal få den til å lagre informasjonen man skriver inn
  const [name, setName] = useState("");
  const [companyName, setCompanyName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");
  const toast = useToast();

  return (
    <form>
      <div className="formInfo">
        <TextField
          required
          color={"primary"}
          label={"Name"}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <PersonIcon color={"primary"} />
              </InputAdornment>
            ),
          }}
          variant={"standard"}
        />
      </div>
      <div className="formInfo">
        <TextField
          required
          label={"Co. Name"}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <BusinessIcon color={"primary"} />
              </InputAdornment>
            ),
          }}
          variant={"standard"}
        />
      </div>
      <div className="formInfo">
        <TextField
          required
          label={"Email"}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <EmailIcon color={"primary"} />
              </InputAdornment>
            ),
          }}
          variant={"standard"}
        />
      </div>
      <div className="formInfo">
        <TextField required label="Message" multiline rows={4} />
      </div>
      <div align="center">
        <CustomButton
          variant="contained"
          to={"/contact"}
          onClick={() =>
            toast({
              title: "Message sent.",
              description: "Thank you for reaching out to us.",
              status: "success",
              duration: 9000,
              isClosable: true,
            })
          }
        >
          Submit
        </CustomButton>
      </div>
    </form>
  );
}
