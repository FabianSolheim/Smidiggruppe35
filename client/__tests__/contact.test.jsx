/*import React from "react";
import ReactDOM from "react-dom";
import { Contact } from "../src/pages/Contact/contact";

describe("Contact page", () => {
  it("shows contact page", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(<Contact />, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});*/
