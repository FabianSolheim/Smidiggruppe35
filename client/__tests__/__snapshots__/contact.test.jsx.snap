// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact page shows contact page 1`] = `"<div class=\\"MuiBox-root css-1ctseqc\\"><div class=\\"MuiGrid-root MuiGrid-container css-1hbmzt3-MuiGrid-root\\"><header class=\\"contactHeader\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root\\"><div><div style=\\"font-size: 36px; font-weight: 500; text-align: center; color: rgb(33, 33, 33); font-family: Josefin Sans;\\">Contact</div></div><div style=\\"width: 120px; height: 2px; background-color: rgb(73, 3, 116); margin: 5px auto 20px auto;\\"></div><p style=\\"max-width: 300px; text-align: center; margin-bottom: 30px;\\">Reach out if you want to <span class=\\"textHighlight\\">collaborate</span> or know more about our services, or just to <span class=\\"textHighlight\\">connect</span>.</p></div></header><div class=\\"MuiGrid-root MuiGrid-container css-1wifiq9-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><form><div class=\\"formInfo\\"><div class=\\"MuiFormControl-root MuiTextField-root css-15fgmos-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"mui-1\\" id=\\"mui-1-label\\">Contact person<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" required=\\"\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"\\" id=\\"mui-1\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1q8v7ak-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"PersonIcon\\"><path d=\\"M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z\\"></path></svg></div></div></div></div><div class=\\"formInfo\\"><div class=\\"MuiFormControl-root MuiTextField-root css-15fgmos-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"mui-2\\" id=\\"mui-2-label\\">Co. Name<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" required=\\"\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"\\" id=\\"mui-2\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1q8v7ak-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"BusinessIcon\\"><path d=\\"M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z\\"></path></svg></div></div></div></div><div class=\\"formInfo\\"><div class=\\"MuiFormControl-root MuiTextField-root css-15fgmos-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"mui-3\\" id=\\"mui-3-label\\">Email<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" required=\\"\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"\\" id=\\"mui-3\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1q8v7ak-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"EmailIcon\\"><path d=\\"M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z\\"></path></svg></div></div></div></div><div class=\\"formInfo\\"><div class=\\"MuiFormControl-root MuiTextField-root css-15fgmos-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"mui-4\\" id=\\"mui-4-label\\">Message<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root\\"><textarea rows=\\"4\\" aria-invalid=\\"false\\" required=\\"\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input\\" id=\\"mui-4\\" style=\\"height: 0px; overflow: hidden;\\"></textarea><textarea aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input\\" readonly=\\"\\" tabindex=\\"-1\\" style=\\"visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;\\"></textarea><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Message&nbsp;*</span></legend></fieldset></div></div></div><div align=\\"center\\"><button class=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"submit\\" to=\\"\\">Submit<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></form></div><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><div class=\\"contactInfo\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-vpg1h7-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"PhoneIcon\\"><path d=\\"M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z\\"></path></svg><p><a style=\\"color: black;\\" href=\\"tel:+4792452610\\" class=\\"contactInfoHover\\">+47 924 52 610</a></p></div><div class=\\"contactInfo\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-vpg1h7-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"EmailIcon\\"><path d=\\"M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z\\"></path></svg><p><a class=\\"contactInfoHover\\" style=\\"color: black;\\" href=\\"mailto:contact@melioraimpact.no\\">contact@melioraimpact.no</a></p></div><div class=\\"contactInfo\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-vpg1h7-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"LocationOnIcon\\"><path d=\\"M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z\\"></path></svg><p>Peder Ankers plass 49</p><p>0301 Oslo</p></div></div></div></div></div>"`;
